engine/execution/state/cargo/queue/errors.go

```diff
@@ -0,0 +1,35 @@
+package queue
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// QueueCapacityReachedError is returned when finalized block queue has reached its capacity
+type QueueCapacityReachedError struct {
+	Capacity int
+}
+
+func (e QueueCapacityReachedError) Error() string {
+	return fmt.Sprintf("finalized block queue has reached its capacity: %d", e.Capacity)
+}
+
+// NonCompliantHeaderError is returned when a new header is pushed to the queue
+// that is not compliant with the previously added header because of height or parent id mismatch
+type NonCompliantHeaderError struct {
+	ExpectedBlockHeight, ReceivedBlockHeight     uint64
```

Comment:
could you avoid using "list"-style type definition, and explicitly specify type for each field (/ argument in method signature)?  the list-style type definition makes search and replace difficult.

Reply:
updated at [8aeec60](https://github.com/onflow/flow-go/pull/4246/commits/8aeec602378e7206772b636f40c978612b1d2007)